src/test/java/de/tobiasroeser/lambdatest/proxy/ExampleProxyTest.java

```diff
@@ -107,22 +118,79 @@ public boolean genericArg(String arg1) {
 				}
 			});
 			intercept(UnsupportedOperationException.class,
-					"(?s).*\\Qpublic boolean genericArg(Object object1)\\E.*",
+					"(?s).*\\Qpublic boolean genericArg(T x0)\\E.*",
 					() -> dep.genericArg("1"));
 		});
-		test("A proxy with missing implementation should print a nice (copy 'n paste -able) method signature",
-				() -> {
-					final Dependency dep = TestProxy.proxy(Dependency.class);
-					intercept(UnsupportedOperationException.class,
-							"(?s).*\\Qpublic int foobar(String string1, int int1)\\E.*",
-							() -> dep.foobar("a", 1));
-
-					intercept(UnsupportedOperationException.class,
-							"(?s).*\\Qpublic Object baz(List list1)\\E.*",
-							() -> dep.baz(Arrays.asList("1")));
 
+    section("A proxy with with missing implementation should print a nice (copy 'n paste -able) method signature",
+				() -> {
+					test("case: int foobar(String s1, int i2)",
+							() -> {
+								final Dependency dep = TestProxy.proxy(Dependency.class);
+                intercept(UnsupportedOperationException.class,
+										"(?s).*public int foobar\\(String x0 ,int x1\\).*",
+										() -> dep.foobar("a", 1));
+							});
+					test("case: U baz(List<U> arg1)",
+							() -> {
+								final Dependency dep = TestProxy.proxy(Dependency.class);
+								intercept(UnsupportedOperationException.class,
+										"(?s).*public U baz\\(List<U> x0\\).*",
+										() -> dep.baz(new ArrayList()));
+							});
+					test("case: String list(List<String> strings)",
+							() -> {
+								final Dependency dep = TestProxy.proxy(Dependency.class, new Object() {
+									@SuppressWarnings("unused")
+									public String hello() {
+										return "Hello Proxy!";
+									}
+								});
+								intercept(UnsupportedOperationException.class,
+										"(?s).*public String list\\(List<String> x0\\).*",
+										() -> dep.list(new ArrayList<>()));
+							});
+
+					test("case: <T> T generics1(List<T> ts)",
+							() -> {
+								final Dependency dep = TestProxy.proxy(Dependency.class, new Object() {
+									@SuppressWarnings("unused")
+									public String hello() {
+										return "Hello Proxy!";
+									}
+								});
+								intercept(UnsupportedOperationException.class,
+										"(?s).*public <T> T generics1\\(List<T> x0\\).*",
```

Comment:
All signatures with a `T` or any other unbound type variable aren't exactly copy-n-paste friendly. The programmer has to replace them with concrete types,  thus it would be more useful to suggest a `Object` for the type parameter or the raw type. Same in following cases.